platform/docs/docs/platform/services/ui/ui-customization-service.md

```diff
@@ -0,0 +1,196 @@
+---
+sidebar_position: 3
+sidebar_label: UI Modal Service
+---
+# UI Customization Service
+
+There are a lot of places where users may want to configure certain elements
+differently between different modes or for different deployments.  A mode
+example might be the ability to select a different right click context menu
+for the cornerstone extension.  A deployment example might be the ability to add
+a selector for the data source back end to use to the study results table.
+
+The use of this service enables these to be defined in a typed fashion by
+providing an easy way to set default values for this, but to allow a non-default value to be specified by the configuration or mode.
+
+This service is a UI service in that part of the registration allows for registering
+UI components and types to deal with, but it does not directly provide an UI
+displayable elements unless customized to do so.
+
+## Registering Customizations
+There are several ways to register customizations.  The `APP_CONFIG.whiteLabeling`
+field is used as a per-configuration entry.  This object can list single
+configurations by id, or it can list sets of customizations by referring to
+the `customizationModule` in an extension.
+
+As well, extensions can register default customizations by provided a 'default'
+name key within the extension.
+
+Mode specific customizations can be registered in the `onModeEnter` of the mode.
+Note that mode specific customizations are CLEARED before this call is made
+by a call to the `uiCustomizationService.onModeEnter` from the `ExtensionManager`.
+
+The following example shows first the registration of the default values,
+in this case they are base types which are inheritted later:
+
+```js
+// In the cornerstone extension  getCustomizationModule:
+const getCustomizationModule = () => ([
+  {
+    name: 'default',
+    value: [
+      {
+        id: 'ohif.cornerstoneContextMenu',
+        commands: [... default commands for the context menu],
+        content: CornerstoneOverlay,
+        contentProps: ....
+      },
+      {
+        id: 'ohif.menuItem',
+        content: CornerstoneMenuItem,
+      },
+    ],
+  },
+]);
+```
+
+Then, in the configuration file one might have a custom overlay definition:
+
+```js
+// in the APP_CONFIG file set the top right area to show the patient name
+// using PN: as a prefix when the study has a non-empty patient name.
+whiteLabeling: {
```

Comment:
Above, you explained by contextMenu as default registration, and here you are switching to the overlay top right. This will be better recieved if above also you show the default reigstration for the topRight.